server/routes/handlers/book/issuesNeeds.ts

```diff
@@ -0,0 +1,91 @@
+import { Request, Response } from 'express'
+import { updateRecall } from '../../../clients/manageRecallsApi/manageRecallsApiClient'
+import logger from '../../../../logger'
+import { isInvalid, makeErrorObject } from '../helpers'
+
+export const issuesNeeds = async (req: Request, res: Response): Promise<void> => {
+  const { nomsNumber, recallId } = req.params
+  if (isInvalid(nomsNumber) || isInvalid(recallId)) {
+    return res.redirect(303, `/persons/${nomsNumber}`)
+  }
+
+  const { contraband, contrabandDetail, vulnerabilityDiversity, vulnerabilityDiversityDetail, mappaLevel } = req.body
+
+  if (
+    !contraband ||
+    !vulnerabilityDiversity ||
+    (contraband === 'true' && !contrabandDetail) ||
```

Comment:
Minor but because values always arrive as strings not booleans it can confuse less if they're 'yes' / 'no' rather than 'true' / 'false' (both of the latter are always truthy so tests for them need to be as exact as above)